File: Programming Language Performance Tests/Kotlin/INI_Performance_Test.kt
import java.io.FileReader
import java.io.FileWriter
import java.lang.Exception
import org.ini4j.Ini

class INI_Performance_Test {
    data class IniEntry(var SECTION: String, var KEY: String, var VALUE: String)

    fun execute() {
        val iniEntries = mutableListOf<IniEntry>()
        try {
            Ini(FileReader("performance_test.ini")).run {
                val sections = keys
                for (sectionName in sections) {
                    for (i in 1..5) {
                        iniEntries += IniEntry(sectionName, sectionName + i, get(sectionName, sectionName + i))
                    }
                    val lastEntry = iniEntries.last()
                    put(lastEntry.SECTION, lastEntry.KEY, "Testing around")
                    store(FileWriter("performance_test.ini"))
                    put(lastEntry.SECTION, lastEntry.KEY, lastEntry.VALUE)
                    store(FileWriter("performance_test.ini"))
                }
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}

File: Programming Language Performance Tests/Scala/INI_Performance_Test.scala
import org.ini4j.Ini

import java.io.{FileReader, FileWriter, IOException}
import scala.collection.mutable.ListBuffer

class INI_Performance_Test {
  class IniEntry(private var SECTION: String, private var KEY: String, private var VALUE: String){
    def getSECTION: String = SECTION

    def setSECTION(SECTION: String): Unit = {
      this.SECTION = SECTION
    }

    def getKEY: String = KEY

    def setKEY(KEY: String): Unit = {
      this.KEY = KEY
    }

    def getVALUE: String = VALUE

    def setVALUE(VALUE: String): Unit = {
      this.VALUE = VALUE
    }
  }

  def execute(): Unit = {
    val iniEntries = new ListBuffer[IniEntry]
    try {
      val ini: Ini = new Ini(new FileReader("performance_test.ini"))
      val sections = ini.keySet()
      sections.forEach(sectionName => {
        for (i <- 1 until 5){
          iniEntries += new IniEntry(sectionName, sectionName + i, ini.get(sectionName, sectionName + i))
        }
        val lastEntry = iniEntries.last
        ini.put(lastEntry.getSECTION, lastEntry.getKEY, "Testing around")
        ini.store(new FileWriter("performance_test.ini"))
        ini.put(lastEntry.getSECTION, lastEntry.getKEY, lastEntry.getVALUE)
        ini.store(new FileWriter("performance_test.ini"))
      })
    }
    catch {
      case e: IOException => e.printStackTrace()
    }
  }
}
